### PR TITLE
examples/go: Add skeleton for Go hooks

### PR DESCRIPTION
**Pull request by enisoc**
This controller doesn't do anything useful. It's just an example skeleton for writing Metacontroller hooks with Go.

### CODE REVIEW

File: examples/go/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.10 AS build
```
Comment (luisdavim) [inline examples/go/Dockerfile]: Can you use `golang:1.11`?

File: examples/go/thing-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    resource: things
+  childResources:
+    - apiVersion: v1
+      resource: pods
+  hooks:
+    sync:
+      webhook:
+        url: http://thing-controller.metacontroller/sync
+---
+apiVersion: apps/v1beta1
```
Comment (luisdavim) [inline examples/go/thing-controller.yaml]: Can you use `apps/v1`?

**Comment by enisoc** [comment]
I posted this as an example, but I don't think it's ready to merge. Given the problems with generating JSON from statically-typed languages (#76), I'd like to limit the examples to dynamic languages for now.

**Comment by luisdavim** [comment]
I understand but you could just write that in a disclaimer in the readme and make it available for people that want to use go.

**Comment by enisoc** [comment]
Alright that's a good point. I'll clean it up and add the disclaimer as you suggested.

**Pull request merged**